Revises abstract methods in `TransitionCapableStateMachineTrait`

mecha opened the pull request:
Resolves #6

mecha committed:
Renamed abstract `_getStateMachine()` method

Part 1 of the solution for #6

mecha committed:
Abstract exception methods now throw methods

Part 2 of the solution for #6

mecha committed:
Updated changelog

Related to #6

XedinUnknown reviewed (commented):
Don't you think that the names need to be more specific, such as `*ForTransition()`?

mecha commented:
Sure.

mecha committed:
Renamed abstract method to be more specific ...

From `_getStateMachineFor()` to `_getStateMachineForTransition()`.

Part of the solution for #6
